patches/tModLoader/Terraria/ModLoader/ModAccessorySlot.cs

```diff
@@ -0,0 +1,86 @@
+using Microsoft.Xna.Framework;
+using Terraria.ModLoader.Default;
+using Terraria.UI;
+
+//NOTE: Does not fully support: ItemLoader.GetGamepadInstructions and related gamepad stuff.
+//TODO: Documentation?
+
+namespace Terraria.ModLoader
+{
+	/// <summary>
+	/// A ModAccessorySlot instance represents a net new accessory slot instance. You can store fields in the ModAccessorySlot class. 
+	/// </summary>
+	public abstract class ModAccessorySlot : ModType
+	{
+		internal int type;
+		internal int index;
+		public AccessorySlotLoader Loader => LoaderManager.Get<AccessorySlotLoader>();
+
+		// Properties to preset a location for the accessory slot
+		public virtual int XLoc => 0;
+		public virtual int YLoc => 0;
+
+		// Properties to change the Default Background Texture
+		public virtual string DyeBackgroundTexture => null;
+		public virtual string VanityBackgroundTexture => null;
+		public virtual string FunctionalBackgroundTexture => null;
+
+		// Properties to change the Default Style Image Texture
+		public virtual string DyeTexture => null;
+		public virtual string VanityTexture => null;
+		public virtual string FunctionalTexture => null;
+
+		// Properties to control which parts of accessory slot draw
+		public virtual bool DrawFunctionalSlot => true;
+		public virtual bool DrawVanitySlot => true;
+		public virtual bool DrawDyeSlot => true;
+
+		protected sealed override void Register() {
+			type = Loader.Register(this);
+		}
+
+		private bool MySlotContainsAnItem() {
```

Comment:
Suggestion, invert this and call it `IsEmpty`
Can use the suggested properties above.
Should probably also check dye slot.